### PR TITLE
Fix networking issues

### PR DESCRIPTION
**Pull request by rachelg4**
- Add healthcheck at /heatbeat
- Add basic auth to elasticsearch and remove from kibana
- Set up ssl for kibana only

### CODE REVIEW

**Comment by rachelg4** [comment]
@stvstnfrd

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2 [hunk 1]
```diff
@@ -65,5 +66,15 @@ server {
 	  if (-f $request_filename) {
 		  break;
 	  }
+
+  # Health Check
+  location /heartbeat {
+          rewrite /heartbeat / break;
```
Comment (stvstnfrd) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2]: I think this should rewrite to `/e/`?

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2 [hunk 2]
```diff
@@ -65,5 +66,15 @@ server {
...
+          proxy_set_header X-Real-IP $remote_addr;
+          proxy_set_header X-Forwarded-For $http_x_forwarded_for;
+
```
Comment (stvstnfrd) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2]: Nit: kill superfluous newline

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2 (hunk 1 above)
Comment (stvstnfrd) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2]: oh, you're passing it off to the ES upstream?

File: playbooks/roles/nginx/defaults/main.yml [hunk 3]
```diff
@@ -12,6 +12,7 @@ NGINX_EDXAPP_CUSTOM_REDIRECTS: {}
 NGINX_ELASTICSEARCH_HOST: localhost
 
 NGINX_ENABLE_SSL: False
+NGINX_KIBANA_ENABLE_SSL: False
```
Comment (stvstnfrd) [inline playbooks/roles/nginx/defaults/main.yml]: This should actually be called `KIBANA_ENABLE_SSL` and it should be included in the kibana role defaults.

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2 (hunk 1 above)
Comment (rachelg4) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/kibana.j2]: Yes it is passed to upstream

**Pull request merged**